tests/js/server/aql/aql-queries-optimizer-count.js

```diff
@@ -0,0 +1,105 @@
+/*jshint globalstrict:false, strict:false, maxlen: 500 */
+/*global assertEqual, AQL_EXPLAIN */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief tests for query language, sort optimizations
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Heiko Kernbach
+/// @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+let jsunity = require("jsunity");
+let internal = require("internal");
+let helper = require("@arangodb/aql-helper");
+let getQueryResults = helper.getQueryResults;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function ahuacatlQueryOptimizerCollectTestSuite() {
+  let collection = null;
+  let cn = "UnitTestsAhuacatlOptimizerCountCollect";
+
+  return {
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief set up
+////////////////////////////////////////////////////////////////////////////////
+
+    setUp: function () {
+      internal.db._drop(cn);
+      collection = internal.db._create(cn);
+
+      // create an index with two attributes (first: boolean, second: array)
+      collection.ensureIndex({type: "persistent", fields: ["someBoolean", "someArray[*]"]});
+
+      // create 100 (50+50) documents in total
+      for (let outer = 0; outer < 50; outer++) {
+        let myArray = [];
+        // create 100 entries for our inner document array
+        for (let inner = 0; inner < 10; inner++) {
+          // write numeric strings
+          myArray.push(JSON.stringify(inner));
+        }
+        collection.save({"someBoolean": true, "someArray": myArray});
+        collection.save({"someBoolean": false, "someArray": myArray});
```

Comment:
Better save all documents in an array, and insert all at once. That saves time when running the tests.